tensorflow/python/keras/layers/advanced_activations.py

```diff
@@ -333,3 +333,63 @@ def get_config(self):
   @tf_utils.shape_type_conversion
   def compute_output_shape(self, input_shape):
     return input_shape
+
+
+@keras_export('keras.layers.RReLU')
+class RReLU(Layer):
+  """Randomized Leaky Rectified Liner Unit function
+  It Randomizes the alpha passed to the leaky Relu function
+
+  'f(x) = alpha * x for x <0',
+  'f(x) = x for x>=0'.
+
+  where alpha is randomly sampled from uniform distribution U(lower,upper).
+
+  See: https://arxiv.org/pdf/1505.00853.pdf
+
+  Input shape:
+      Arbitrary. Use the keyword argument `input_shape`
+      (tuple of integers, does not include the samples axis)
+      when using this layer as the first layer in a model.
+
+  Output shape:
+      Same shape as the input.
+
+  Arguments:
+      lower: lower bound of the uniform distribution.
+      upper: upper bound of the uniform distribution.
+      seed: Used to create random seeds.
+  """
+
+  def __init__(self, lower=0.125, upper=0.333, seed=None, **kwargs):
+    super(RReLU, self).__init__(**kwargs)
+    if lower < 0 or upper < 0  :
+      raise ValueError('lower/upper bound for RandomizedRectifierLayer needs '
+                       'to be > 0. ')
+    if lower > upper :
+      raise ValueError('Upper bound for RandomizedRectifierLayer needs '
+                       'to be higher than lower bound. ')
+
+    self.lower = K.cast_to_floatx(lower)
+    self.upper = K.cast_to_floatx(upper)
```

Comment:
Can you check shape equivalence?

Reply:
These are scalars and type check might be redundant.